testing/util/deneb.go

```diff
@@ -67,22 +68,48 @@ func GenerateTestDenebBlockWithSidecar(t *testing.T, parent [32]byte, slot primi
 	root, err := block.Block.HashTreeRoot()
 	require.NoError(t, err)
 
-	sidecars := make([]*ethpb.DeprecatedBlobSidecar, len(commitments))
+	sidecars := make([]blocks.ROBlob, len(commitments))
+	sbb, err := blocks.NewSignedBeaconBlock(block)
+	require.NoError(t, err)
+	sh, err := interfaces.SignedBeaconBlockHeaderFromBlockInterface(sbb)
+	require.NoError(t, err)
 	for i, c := range block.Block.Body.BlobKzgCommitments {
-		sidecars[i] = GenerateTestDenebBlobSidecar(root, block, i, c)
+		sidecars[i] = GenerateTestDenebBlobSidecar(t, root, sh, i, c)
 	}
 
-	sbb, err := blocks.NewSignedBeaconBlock(block)
-	require.NoError(t, err)
 	rob, err := blocks.NewROBlock(sbb)
 	require.NoError(t, err)
 	return rob, sidecars
 }
 
-func GenerateTestDenebBlobSidecar(root [32]byte, block *ethpb.SignedBeaconBlockDeneb, index int, commitment []byte) *ethpb.DeprecatedBlobSidecar {
+func GenerateTestDenebBlobSidecar(t *testing.T, root [32]byte, header *ethpb.SignedBeaconBlockHeader, index int, commitment []byte) blocks.ROBlob {
+	blob := make([]byte, fieldparams.BlobSize)
+	binary.LittleEndian.PutUint64(blob, uint64(index))
+	pb := &ethpb.BlobSidecar{
+		SignedBlockHeader: header,
+		Index:             uint64(index),
+		Blob:              blob,
+		KzgCommitment:     commitment,
+		KzgProof:          commitment,
+	}
+	pb.CommitmentInclusionProof = fakeEmptyProof(t, pb)
+	r, err := blocks.NewROBlobWithRoot(pb, root)
+	require.NoError(t, err)
+	return r
+}
+
+func fakeEmptyProof(_ *testing.T, _ *ethpb.BlobSidecar) [][]byte {
+	r := make([][]byte, 17)
```

Comment:
Maybe 17 should be config in fieldparams. Sorry I didn't define that previously

Reply:
My thinking was we'd generate real proofs as the pieces of this puzzle come together and throw this fakeEmptyProof helper away.

Reply:
I was going to say it feels goofy adding this since it's more an empirical fact than a config value, but it turns out the spec does treat it as a [Preset](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/p2p-interface.md#preset), so I'll go ahead and add it.

Reply:
https://github.com/prysmaticlabs/prysm/pull/13190/commits/ccf828f06ec2aced2d4695d3cd2806e60391e5e9